src/celeritas/mat/MaterialParams.cc

```diff
@@ -238,6 +313,31 @@ void MaterialParams::append_element_def(ElementInput const& inp,
     result.atomic_number = inp.atomic_number;
     result.atomic_mass = inp.atomic_mass;
 
+    if (isotope_data_ && !inp.isotopes_fractions.empty())
+    {
+        // There is isotopic data and this specific element has isotopes loaded
+        std::vector<ElIsotopeComponent> vec_eic;
+        for (auto const& key : inp.isotopes_fractions)
+        {
+            vec_eic.push_back(ElIsotopeComponent{key.first, key.second});
+        }
+
+        // Sort isotopes by increasing isotope ID for improved access
+        std::sort(
+            vec_eic.begin(),
+            vec_eic.end(),
+            [](ElIsotopeComponent const& lhs, ElIsotopeComponent const& rhs) {
+                return lhs.isotope < rhs.isotope;
+            });
+
+        result.isotopes = make_builder(&host_data->isocomponents)
+                              .insert_back(vec_eic.begin(), vec_eic.end());
+
+        // Update maximum number of isotopes
+        host_data->max_isotope_components = std::max(
+            host_data->max_isotope_components, result.isotopes.size());
+    }
```

Comment:
I don't think the conditional is needed here either. All of the operations will be null-op if it's empty.